Sources/Core/GTMSessionFetcherLogging.m

```diff
@@ -75,141 +77,165 @@ @implementation GTMSessionFetcher (GTMSessionFetcherLogging)
-                                              attributes:nil
-                                                   error:NULL];
         if (doesFolderExist) {
-          // The directory has been created. Exclude it from backups.
-          NSURL *pathURL = [NSURL fileURLWithPath:logsFolderPath isDirectory:YES];
-          [pathURL setResourceValue:@YES forKey:NSURLIsExcludedFromBackupKey error:NULL];
+          // it's there; store it in the global
+          gLoggingDirectoryPath = [logsFolderPath copy];
         }
       }
-
-      if (doesFolderExist) {
-        // it's there; store it in the global
-        gLoggingDirectoryPath = [logsFolderPath copy];
-      }
     }
+    return gLoggingDirectoryPath;
   }
-  return gLoggingDirectoryPath;
 }
 
 + (void)setLogDirectoryForCurrentRun:(NSString *)logDirectoryForCurrentRun {
-  // Set the path for this run's logs.
-  gLogDirectoryForCurrentRun = [logDirectoryForCurrentRun copy];
+  @synchronized([GTMSessionFetcher class]) {
+    // Set the path for this run's logs.
+    gLogDirectoryForCurrentRun = [logDirectoryForCurrentRun copy];
+  }
 }
 
 + (NSString *)logDirectoryForCurrentRun {
-  // make a directory for this run's logs, like SyncProto_logs_10-16_01-56-58PM
-  if (gLogDirectoryForCurrentRun) return gLogDirectoryForCurrentRun;
+  @synchronized([GTMSessionFetcher class]) {
+    // make a directory for this run's logs, like SyncProto_logs_10-16_01-56-58PM
+    if (gLogDirectoryForCurrentRun) return gLogDirectoryForCurrentRun;
 
-  NSString *parentDir = [self loggingDirectory];
-  NSString *logNamePrefix = [self processNameLogPrefix];
-  NSString *dateStamp = [self loggingDateStamp];
-  NSString *dirName = [NSString stringWithFormat:@"%@%@", logNamePrefix, dateStamp];
-  NSString *logDirectory = [parentDir stringByAppendingPathComponent:dirName];
+    NSString *parentDir = [self loggingDirectory];
+    NSString *logNamePrefix = [self processNameLogPrefix];
+    NSString *dateStamp = [self loggingDateStamp];
+    NSString *dirName = [NSString stringWithFormat:@"%@%@", logNamePrefix, dateStamp];
+    NSString *logDirectory = [parentDir stringByAppendingPathComponent:dirName];
 
-  if (gIsLoggingToFile) {
-    NSFileManager *fileMgr = [NSFileManager defaultManager];
-    // Be sure that the first time this app runs, it's not writing to a preexisting folder
-    static BOOL gShouldReuseFolder = NO;
-    if (!gShouldReuseFolder) {
-      gShouldReuseFolder = YES;
-      NSString *origLogDir = logDirectory;
-      for (int ctr = 2; ctr < 20; ++ctr) {
-        if (![fileMgr fileExistsAtPath:logDirectory]) break;
-
-        // append a digit
-        logDirectory = [origLogDir stringByAppendingFormat:@"_%d", ctr];
+    if (gIsLoggingToFile) {
+      NSFileManager *fileMgr = [NSFileManager defaultManager];
+      // Be sure that the first time this app runs, it's not writing to a preexisting folder
+      static BOOL gShouldReuseFolder = NO;
```

Comment:
Do you have another thread sanitizer issue here with an unprotected global?

Reply:
How?  This whole body is within a sync block and only use here.